index.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const NR = require('node-resque');
+const asCallback = require('ascallback');
+const connectionDetails = {
+    pkg: 'ioredis',
+    host: process.env.REDIS_HOST,
+    password: process.env.REDIS_PASSWORD,
+    port: process.env.REDIS_PORT,
+    database: 0
+};
+
+/**
+ * Construct an executor and call executor.start with the buildConfig
+ * @method execute
+ * @param  {Object}     config                      config passed in from executor-queue
+ * @param  {Object}     config.executor             config for the executor
+ * @param  {String}     config.executor.name        executor name
+ * @param  {Object}     config.executor.options     options to pass into executor's constructor
+ * @param  {Object}     config.buildConfig          buildConfig needed for executor.start
+ */
+function execute(config) {
+    return new Promise((resolve) => {
+        if (config.executor.name === 'queue') {
+            throw new Error('screwdriver-executor-queue is not a valid option');
+        }
+        // eslint-disable-next-line
+        const ExecutorPlugin = require(`screwdriver-executor-${config.executor.name}`);
+        const executor = new ExecutorPlugin(config.executor.options);
+
+        resolve(executor);
```

Comment:
This was
```
new Promise ((resolve, reject) 
...
```
so we didn't need the `return`

https://stackoverflow.com/questions/26711243/promise-resolve-vs-new-promiseresolve